lib/iris/util.py

```diff
@@ -1706,7 +1706,8 @@ def _meshgrid(*xi, **kwargs):
 def find_discontiguities(cube, rel_tol=1e-5, abs_tol=1e-8):
     """
     Searches coord for discontiguities in the bounds array, returned as a
-    boolean array (True where discontiguities are present).
+    boolean array (True for all cells which are discontiguous with the cell
+    immediately above them or to their right).
```

Comment:
I believe this is a more accurate description of what it **_used_** to do, but is no longer actually correct, since behaviour has changed  ??
I.E. result boolean array will now be True for each cell which is discontiguous with neighbours in any one of 8 different ways (4 directions in each of  2 coordinates).

Reply:
I believe it's still exhibiting the same behaviour except that some discontiguities were not showing up previously. Each *discontiguity* is picked up but it is only associated with the *cell* that is immediately to its left or bellow it. The cell to its right will not be flagged as discontiguous (if there are no other discontiguities). This is a bit arbitrary, but the function has the property that all neighbouring pairs of cells which aren't flagged as discontiguous will have no discontiguities betwen them. The behaviour you are suggesting would have the additional property that all cells not flagged as discontiguous will have *no* discontiguities with *any* neighbouring cells. This may be a useful property which we may want to implement in future, but for the time being I've opted for keeping close to the original design.

Reply:
But I still think that the part 
> `(True for all cells which are discontiguous with the cell immediately above them or to their right)`

is actually incorrect, if you read it as implying that it is _not_ True for a cell which is discontiguous with those _below or to the left_.  But in fact, such cells are now _also_ marked as 'True' -- i.e. it should now say something like 
> `(True for all cells which are discontiguous with any of those immediately above, below, to left or right)`

? I am misunderstanding this ?

Reply:
I think maybe you are misunderstanding this. If you look at the tests, we add a discontiguity to, for example, the two top most bounds of a cell. This makes it discontiguous with the cells to its left and right but if you look at the expected and returned value, there are only two cells flagged in that row, not three.

Reply:
Ok !
@stephenworsley and I have had a long offline discussion + investigation of the behaviour (old + new) of this.
I'm now comfortable that it is doing something more correct, relative to the existing behaviour, and is now better explained.
I'm not so confident _why_ this is the right thing to do, but IIRC the whole thing was invented to address some imagined possible problems with plotting data with 2D bounds via pcolormesh.  What it now does is slightly more "cautious", masking some extra problem points in some cases.  Unfortunately I think we never had a definitive rationale or even test examples to define exactly what is needed here.